internal/cloudapi/v2/handler.go

```diff
@@ -888,6 +891,167 @@ func (h *apiHandlers) getComposeManifestsImpl(ctx echo.Context, id string) error
 	return ctx.JSON(http.StatusOK, resp)
 }
 
+// sbomsFromOSBuildJob extracts SBOM documents from dependencies of an OSBuild job.
+func sbomsFromOSBuildJob(w *worker.Server, osbuildJobUUID uuid.UUID) ([]ImageSBOM, error) {
+	var osbuildJobResult worker.OSBuildJobResult
+	osbuildJobInfo, err := w.OSBuildJobInfo(osbuildJobUUID, &osbuildJobResult)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get results for OSBuild job %q: %v", osbuildJobUUID, err)
+	}
+
+	pipelineNameToPurpose := func(pipelineName string) (ImageSBOMPipelinePurpose, error) {
+		if slices.Contains(osbuildJobResult.PipelineNames.Payload, pipelineName) {
+			return ImageSBOMPipelinePurposeImage, nil
+		}
+		if slices.Contains(osbuildJobResult.PipelineNames.Build, pipelineName) {
+			return ImageSBOMPipelinePurposeBuildroot, nil
+		}
+		return "", fmt.Errorf("Pipeline %q is not listed as either a payload or build pipeline", pipelineName)
+	}
+
+	// SBOMs are attached to the depsolve job results.
+	// Depsolve jobs are dependencies of Manifest job.
+	// Manifest job is a dependency of OSBuild job.
+	manifesJobInfo, _, err := manifestJobResultsFromJobDeps(w, osbuildJobInfo.Deps)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get manifest job info for OSBuild job %q: %v", osbuildJobUUID, err)
+	}
+
+	var imageSBOMs []ImageSBOM
+	for _, manifestDepUUID := range manifesJobInfo.Deps {
+		depJobType, err := w.JobType(manifestDepUUID)
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get job type for dependency %q: %v", manifestDepUUID, err)
+		}
+
+		if depJobType == worker.JobTypeDepsolve {
```

Comment:
NITPICK: Flipping the condition with `continue` would decrease the indentation for the largest part of the loop body, which is nicer IMO.

Reply:
Yeah, I don't see any problem with it.